### PR TITLE
PR for code review #2

### PR DESCRIPTION
**Pull request by buddygalletti**
Get Floppy with it

### CODE REVIEW

File: backend/routes/products.js [hunk 1]
```diff
@@ -34,16 +34,29 @@ router.get('/category/:id', async (req, res, next) => {
...
   try {
-    const newProduct = await Product.create(req.body);
+    const newProduct = await Product.create({
+      name: req.body.name,
+      imageUrl: req.body.imageUrl,
+      price: req.body.price,
+      brand: req.body.brand,
+      quantity: req.body.quantity,
+      description: req.body.description,
+    });
```
Comment (jdmann) [inline backend/routes/products.js]: consider protecting certain routes for admins only .  this pattern for middleware may be useful for certain routes: ('/', isAdmin, () =>{})

File: backend/routes/products.js [hunk 2]
```diff
@@ -34,16 +34,29 @@ router.get('/category/:id', async (req, res, next) => {
...
     res.status(201).send(newProduct);
   } catch (ex) {
     next(ex);
   }
 });
 
-router.put('/', async (req, res, next) => {
+router.put('/:id', async (req, res, next) => {
```
Comment (wallacepreston) [inline backend/routes/products.js]: Add authentication to your routes.

File: backend/server.js [hunk 3]
```diff
@@ -49,7 +52,12 @@ app.get('/', (req, res, next) => {
...
   try {
     if (req.session.email) {
-      res.send(req.session.email);
+      const user = await User.findOne({
+        where: {
+          email: req.session.email,
+        }
+      })
+      res.send(user);
     }
```
Comment (jdmann) [inline backend/server.js]: add an else{} and res.send an response if session does not have an email - will break if route is called

File: backend/server.js [hunk 4]
```diff
@@ -33,11 +34,13 @@ app.use(
...
+app.use(express.static(distPath));
 app.use('/api/products', require('./routes/products'));
 app.use('/api/users', require('./routes/users'));
 app.use('/api/categories', require('./routes/categories'));
 app.use('/api/cart', require('./routes/cart'));
 app.use('/google', require('./routes/oauth'));
+app.use('/stripe', require('./routes/stripe'));
 
 app.get('/', (req, res, next) => {
   if (req.user) {
```
Comment (jdmann) [inline backend/server.js]: consider moving data from google dataValues directly to req.user

File: backend/specs/authSpec.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    await db.sync({ force: true});
+    await User.create({
+      firstName: 'Auth',
+      lastName: 'Tester',
+      email: 'testAuth@spec.com',
+      password: 'password'
+    });
+  });
+  describe('Login', () => {
+    it('Login with correct credentials', () => {
```
Comment (wallacepreston) [inline backend/specs/authSpec.js]: Like it.  Good job!

File: backend/specs/userSpec.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        })
+        .expect(201)
+        .then(res => {
+          expect(res.body.firstName).to.equal("Testy")
+          expect(res.body.lastName).to.equal('Specy')
+          expect(res.body.email).to.equal('testy@specy.com')
+          expect(res.body.password).to.equal(saltHash('password'))
+        })
+    });
+    it('ALL EMAILS SHOULD BE UNIQUE', () => {
```
Comment (wallacepreston) [inline backend/specs/userSpec.js]: Really cool. :)

File: index.html [hunk 7]
```diff
@@ -1,15 +1,45 @@
...
+      integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo"
+      crossorigin="anonymous"
+    ></script>
+    <script
+      defer
+      src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js"
+      integrity="sha384-UO2eT0CpHqdSJQ6hJty5KVphtPhzWj9WO1clHTMGa3JDZwrnQq4sF86dIHNDz0W1"
+      crossorigin="anonymous"
+    ></script>
+    <script
```
Comment (wallacepreston) [inline index.html]: Suggestion (not required): Download the stylesheets and use it on your own server.

File: package.json [hunk 8]
```diff
@@ -9,8 +9,8 @@
     "webpack:dev": "npm run webpack -- --watch --mode=development",
     "start:dev": "npm run webpack:dev & nodemon ./backend/app.js --ignore dist --ignore src",
     "start": "node ./backend/app.js",
-    "seed": "node ./backend/seed.js",
```
Comment (wallacepreston) [inline package.json]: Cool.  I think you moved this to a conditional env var?

File: index.html [hunk 9]
```diff
@@ -1,15 +1,45 @@
...
+      defer
+      src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js"
+      integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM"
+      crossorigin="anonymous"
+    ></script>
```
Comment (jdmann) [inline index.html]: can we make jquery work well with react - not sure

File: package.json [hunk 10]
```diff
@@ -41,28 +41,32 @@
     "@babel/preset-env": "^7.5.5",
     "@babel/preset-react": "^7.0.0",
     "babel-loader": "^8.0.6",
+    "chai": "^4.2.0",
+    "mocha": "^6.2.0",
     "react": "^16.8.6",
     "react-dom": "^16.8.6",
     "react-redux": "^7.1.0",
     "react-router-dom": "^5.0.1",
+    "react-toastify": "^5.3.2",
```
Comment (wallacepreston) [inline package.json]: Fun!

File: src/components/AdminNavbar.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+const AdminNavBar = () => {
```
Comment (wallacepreston) [inline src/components/AdminNavbar.js]: Like the special Admin Navbar!

File: src/components/App.js [hunk 12]
```diff
@@ -8,51 +8,62 @@ import Cart from './Cart';
...
+        <Route path="/products/search/:name" />
         <Route path="/products/category/:id" component={Products} />
         <Route path="/products" exact component={Products} />
         <Route path="/cart" component={Cart} />
         <Route path="/login" component={Login} />
         <Route path="/signup" component={SignUp} />
         <Route path="/checkout" component={Checkout} />
-        <Route path="/products/search/:name" />
+        <Route path="/orderconfirmation" component={OrderConfirmation} />
+        {this.props.loggedInUser.isAdmin && <Route path="/admincp" component={AdminCP} />}
```
Comment (wallacepreston) [inline src/components/App.js]: Good going using the && syntax.  Also, secure your data on the backend!

File: src/store.js [hunk 13]
```diff
@@ -1,33 +1,56 @@
-import { createStore, combineReducers, applyMiddleware } from "redux";
-import thunk from "redux-thunk";
-import axios from "axios";
+import { createStore, combineReducers, applyMiddleware } from 'redux';
```
Comment (jdmann) [inline src/store.js]: break into smaller files?

File: src/store.js [hunk 14]
```diff
@@ -1,33 +1,56 @@
...
+const DELETE_CART = 'DELETE_CART'
+
+//Admin Actions
+const GET_ALL_USERS = 'GET_ALL_USERS'
+const REMOVE_USER = 'REMOVE_USER'
+
+const adminReducer = (state = [], action) => {
+  switch (action.type) {
+    case GET_ALL_USERS:
+      state = [...state, action.users].flat();
```
Comment (wallacepreston) [inline src/store.js]: Nice use of .flat()! But you could also spread in action.users

**Pull request merged**